rust/actyx/swarm/cli/src/main.rs

```diff
@@ -128,6 +152,16 @@ async fn run() -> Result<()> {
                 ipfs_embed::Event::Discovered(peer_id) => Some(Event::Discovered(peer_id)),
                 ipfs_embed::Event::Unreachable(peer_id) => Some(Event::Unreachable(peer_id)),
                 ipfs_embed::Event::Connected(peer_id) => Some(Event::Connected(peer_id)),
+                // NOTE: ipfs_embed::Event::Connected is not always emitted
+                // Therefore ipfs_embed::Event::ConnectionEstablished is used as a fallback
+                // See:
+                //  - https://docs.rs/crate/ipfs-embed/latest/source/src/net/peers.rs#:~:text=self.notify(Event%3A%3AConnected(c.peer_id))%3B
+                //      Connected event is SOMETIMES emitted, that is only when `event.other_is_established == 0`.
+                //  - https://docs.rs/crate/libp2p-swarm/0.41.1/source/src/lib.rs#:~:text=let%20non_banned_established
+                //      other_established_connection_ids - banned_peers
+                //  - https://docs.rs/crate/libp2p-swarm/0.41.1/source/src/lib.rs#:~:text=let%20non_banned_established
+                //      other_established_connection_ids is the
+                ipfs_embed::Event::ConnectionEstablished(peer_id, _) => Some(Event::Connected(peer_id)),
```

Comment:
With the above workaround we shouldn’t need this line and the Connected line as well.

Reply:
Can we comment those lines instead of removing them for future context's sake?

Reply:
sure!

Reply:
comments added